orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -107,6 +108,10 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(T
                     }
                   });
   }
+  // If the rule-based transformer is not empty, it should be included in the custom transformer list below.
```

Comment:
why we need make this change?

Reply:
This piece of code will never be touched for now as currently there is no place calling this function with non-empty transformers_and_rules_to_enable. But even when it's not empty, we should make it consistent with above empty case, that is put rule transformer at the end instead of at the beginning, as some eliminations depend on the constant folding.